### PR TITLE
Added Util function auth_token2dict

### PR DESCRIPTION
**Pull request by iglesu**
The function splits a token into a dictionary object. If the token is invalid returns None.

### CODE REVIEW

File: dl/Util.py [hunk 1]
```diff
@@ -316,6 +316,40 @@ def split_auth_token(token):
     return res.groups() if res else None
 
 
+def auth_token2dict(token):
```
Comment (chaduhduh) [inline dl/Util.py:319]: Consider the name `auth_token_to_dict` or `token_to_dict` (primarily just the `to_dict` portion). My thought is that will make it more consistent with methods in other packages (some classes offer `to_dict` methods.)

File: dl/Util.py [hunk 2]
```diff
@@ -316,6 +316,40 @@ def split_auth_token(token):
...
+              'gid': "numeric group id",
+              'hash': hash
+            }
+    E.g.   { 'username': "testuser",
+              'uid': "3666",
+              'gid': "3666",
+              'hash': "$1$PKCFmMzy$OPpZg/ThBmZe/V8LVPvpi/"
+            }
+            or None if token not valid
+    """
```
Comment (chaduhduh) [inline dl/Util.py:344]: Consider adding tests for this function in `./tests/test_util.py` (assuming there is time)

**Review by chaduhduh — APPROVED** [review]
Hey @iglesu,

This looks good. I left one nitpick comment and one suggestion. Feel free to address those at your discretion but otherwise I think this is all set :raised_hands:

File: dl/Util.py (hunk 2 above)
Comment (iglesu) [reply dl/Util.py:344]: yes, I could add something! thank you for reminding me!

**Pull request merged**